src/utils/task.rs

```diff
@@ -175,6 +273,34 @@ impl ThreadServiceInfo {
     }
 }
 
+impl TokioServiceInfo {
+    /// get the time this service has been running since
+    #[inline]
+    pub fn up_time(&self) -> Duration {
```

Comment:
The common methods of `*ServiceInfo` could be a trait, if it ever needs to be used in a polymorphic way.

Reply:
I'll think about it, since there is only 2 objects I am not sure it is worth adding abstraction here.